assembly/src/ast/mod.rs

```diff
@@ -200,12 +203,26 @@ impl ProgramAst {
     // --------------------------------------------------------------------------------------------
 
     /// Returns byte representation of this [ProgramAst].
-    pub fn to_bytes(&self) -> Vec<u8> {
+    pub fn to_bytes(&self, options: AstSerializationOptions) -> Vec<u8> {
         let mut target = Vec::<u8>::default();
 
+        // serialize the options, so that deserialization knows what to do
+        options.write_into(&mut target);
```

Comment:
I was originally thinking that we'd just pass serialization options to `from_bytes` method and then a higher-level component would keep track of them. But I guess serializing options works too (at least for `ProgramAst`).

Reply:
This just seemed like the safest option, but whether it's the right option depends on whether it's feasible to keep track of the options outside of the serialized program, and whether the extra "header" information in the serialized program is too much overhead.

I don't really have a good sense of the use cases to make this call.

Reply:
Note also that serializing the options is the thing that allows the `SerializableModuleAst` solution to work. If we don't serialize the options, then deserialization using `Deserializable::read_from` would require us to return a `Serializable ModuleAst`, and extracting the embedded `ModuleAst` from that struct gets us into lifetime issues, I think.